repos/system_upgrade/common/actors/opensshpermitrootlogincheck/actor.py

```diff
@@ -87,4 +98,53 @@ def process(self):
-            ] + resources)
+            ] + self.common_resources)
+
+    def process8to9(self, config):
+        # RHEL8 default sshd configuration file is not modified: It will get replaced by rpm and
+        # root will no longer be able to connect through ssh. This will probably result in many
+        # false positives so it will have to be waived a lot
+        if not config.modified:
+            create_report([
+                reporting.Title('Possible problems with remote login using root account'),
+                reporting.Summary(
+                    'OpenSSH configuration file will get updated to RHEL9 '
+                    'version, no longer allowing root login with password. '
+                    'It is a good practice to use non-root administrative '
+                    'user and non-password authentications, but if you rely '
+                    'on the remote root login, this change can lock you out '
+                    'of this system.'
+                ),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags(COMMON_REPORT_TAGS),
+                reporting.Remediation(
+                    hint='If you depend on remote root logins using passwords, '
+                         'consider setting up a different user for remote '
+                         'administration or adding a comment into the '
+                         'sshd_config next to the "PermitRootLogin yes" directive '
+                         'to prevent rpm replacing it during the upgrade.'
+                ),
+                reporting.Flags([reporting.Flags.INHIBITOR])
+            ] + self.common_resources)
+        # If the configuration is modified and contains any directive allowing
+        # root login (which is in default configuration), we are upgrading to
+        # RHEL9 keeping the old "security policy", which might keep the root
+        # login unexpectedly open. This might be just high priority warning
+        if global_value(config, 'prohibit-password') == 'yes':
+            create_report([
+                reporting.Title('Remote root logins globally allowed using password'),
+                reporting.Summary(
+                    'RHEL9 no longer allows remote root logins, but the '
+                    'server configuration explicitly overrides this default. '
+                    'The configuration file will not be updated and root is '
+                    'still going to be allowed to login with password. '
+                    'This is not recommended and considered as a security risk.'
+                ),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags(COMMON_REPORT_TAGS),
+                reporting.Remediation(
+                    hint='If you depend on remote root logins using passwords, '
+                         'consider setting up a different user for remote '
+                         'administration, you can ignore this message.'
```

Comment:
Nit: Maybe make 'You can ignore this message' a new sentence?

Reply:
Thanks for review. Fixing with "Otherwise you can ignore this message." as the original sentence was quite confusing